### PR TITLE
[Admin][Country] Add violation for adding provinces with duplicated codes or names

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12 |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/AddressingBundle/Resources/config/services.xml [hunk 1]
```diff
@@ -62,5 +62,9 @@
         <service id="sylius.validator.zone_cannot_contain_itself" class="Sylius\Bundle\AddressingBundle\Validator\Constraints\ZoneCannotContainItselfValidator">
             <tag name="validator.constraint_validator" alias="sylius_zone_cannot_contain_itself_validator" />
         </service>
+
+        <service id="sylius.validator.unique_province_collection_codes" class="Sylius\Bundle\AddressingBundle\Validator\Constraints\UniqueProvinceCollectionCodesValidator">
+            <tag name="validator.constraint_validator" alias="sylius_unique_province_collection_codes_validator" />
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/AddressingBundle/Resources/config/services.xml]:
```suggestion
            <tag name="validator.constraint_validator" />
```
Alias is not needed, last time I was writing a validator I resigned from using it. I believe it's a left-over from older versions of Symfony.

File: src/Sylius/Bundle/AddressingBundle/Validator/Constraints/UniqueProvinceCollectionCodes.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+use Symfony\Component\Validator\Constraint;
+
+final class UniqueProvinceCollectionCodes extends Constraint
+{
+    public string $message = 'sylius.country.unique_provinces_codes';
+
+    public function validatedBy(): string
+    {
+        return 'sylius_unique_province_collection_codes_validator';
+    }
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/AddressingBundle/Validator/Constraints/UniqueProvinceCollectionCodes.php]: Similarly, as in the validator service's definition – we can get rid of this, and it'll work fine without it.

File: src/Sylius/Bundle/AdminBundle/Resources/views/Country/_form.html.twig [hunk 3]
```diff
@@ -1,5 +1,5 @@
+{{ form_errors(form) }}
 <div class="ui segment">
-    {{ form_errors(form) }}
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Country/_form.html.twig:2]: Why 🤔?

**Review by jakubtobiasz — DISMISSED** [review]

File: src/Sylius/Bundle/AddressingBundle/Resources/config/services.xml (hunk 1 above)
Comment (NoResponseMate) [reply src/Sylius/Bundle/AddressingBundle/Resources/config/services.xml]: It's a difference between explicitly stating the connection constraint-validator vs relying on the magic of `constraintFQCN + 'Validator'`.
I'll leave it as is for now.
We need to discuss ups and downs of each approach, choose one and apply it globally.

File: src/Sylius/Bundle/AdminBundle/Resources/views/Country/_form.html.twig (hunk 3 above)
Comment (NoResponseMate) [reply src/Sylius/Bundle/AdminBundle/Resources/views/Country/_form.html.twig:2]: Why not 🤔?
It contains all errors for the whole form, but gets rendered in a segment containing only `code` and `enabled` fields. Looked weird so I've moved it 🤷

Comment (jakubtobiasz) [reply src/Sylius/Bundle/AdminBundle/Resources/views/Country/_form.html.twig:2]: Just asking 😂

**Comment by SzymonFilipek** [comment]
Quick summary of the state of the province issue:
1. Adding provinces with the same code no longer leads to the server error. Now an appropriate message shows up for the user.
province

2. Adding provinces with different codes but with the same name results in the server error.
province2

**Review by jakubtobiasz — APPROVED** [review]

**Review by TheMilek — APPROVED** [review]

**Review by SzymonFilipek — APPROVED** [review]
Changes passed QA check

**Comment by jakubtobiasz** [comment]
Thank you, @NoResponseMate!

**Pull request merged**